[Test] Regression Test for update_all and apply_edges

VoVAllen opened the pull request:
## Description
<!-- Brief description. Refer to the related issues if existed.
It'll be great if relevant reviewers can be assigned as well.-->

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

## Changes
<!-- You could use following template
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)
-->

jermainewang commented on benchmarks/benchmarks/api/bench_builtin_apply_edges.py:
```diff
@@ -0,0 +1,41 @@
+from .. import utils
+
+
+@utils.benchmark('time', timeout=7200)
```
This is very large. We should have a reasonable time estimation. Does feat_size=512 on livejournal takes too long?

jermainewang commented on benchmarks/benchmarks/api/bench_builtin_apply_edges.py:
```diff
@@ -0,0 +1,41 @@
+
+    reduce_builtin_dict = {
+        'u->e': fn.copy_u('h', 'x'),
+        'v->e': fn.copy_u('h', 'x'),
```
no need to test this

jermainewang commented on benchmarks/benchmarks/api/bench_builtin_apply_edges.py:
```diff
@@ -0,0 +1,41 @@
+        'u->e': fn.copy_u('h', 'x'),
+        'v->e': fn.copy_u('h', 'x'),
+        'u+v': fn.u_add_v('h', 'h', 'x'),
+        'u*v': fn.u_mul_v('h', 'h', 'x'),
```
only test u+v is fine

jermainewang commented on benchmarks/benchmarks/api/bench_builtin_update_all.py:
```diff
@@ -0,0 +1,49 @@
+    
+    msg_builtin_dict = {
+        'copy_u': fn.copy_u('h', 'x'),
+        'copy_v': fn.copy_u('h', 'x'),
```
no need to test this

jermainewang commented:
Some of the tests are redundant. Consider removing them. Also, I'd like to add a graph that is similar to the block graph from neighbor sampling (degrees are similar).

VoVAllen commented:
@jermainewang I addressed the comments and updated the `multi_update_all` tests. I found it will stuck when using asv run, so I added the `thread_wrapped_func`

jermainewang commented on benchmarks/benchmarks/api/bench_builtin_apply_edges.py:
```diff
@@ -0,0 +1,38 @@
+def track_time(graph_name, format, feat_size, reduce_type):
+    device = utils.get_bench_device()
+    graph = utils.get_graph(graph_name, format)
+    # Remove format strict
```
remove format restriction

jermainewang commented on benchmarks/benchmarks/api/bench_builtin_apply_edges.py:
```diff
@@ -0,0 +1,38 @@
+    device = utils.get_bench_device()
+    graph = utils.get_graph(graph_name, format)
+    # Remove format strict
+    graph = graph.formats(['coo', 'csr', 'csc'])
```
you should extend `utils.get_graph` to allow multiple formats

jermainewang commented on benchmarks/benchmarks/api/bench_builtin_multi_update_all.py:
```diff
@@ -0,0 +1,48 @@
+                                                             len(candidate_edges)]
+    graph = dgl.heterograph(dd)
+
+    # Remove format strict
```
the comment doesn't make sense

jermainewang commented on benchmarks/benchmarks/api/bench_builtin_multi_update_all.py:
```diff
@@ -0,0 +1,48 @@
+            fn.copy_src('h', 'm'), fn.sum('m', 'h'))
+    graph.multi_update_all(
+        update_dict,
+        "sum")
```
let's test one more case, "stack" cross-reducer.

jermainewang commented on benchmarks/benchmarks/api/bench_udf_multi_update_all.py:
```diff
@@ -0,0 +1,50 @@
+
+@utils.benchmark('time', timeout=600)
+@utils.parametrize('feat_size', [32, 128, 512])
+@utils.parametrize('num_relations', [3, 6, 12])
```
these configs are quite similar. use `[5, 50, 500]`
